src/Microsoft.ML.Data/Training/MLContext.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using System;
+
+namespace Microsoft.ML
+{
+    /// <summary>
+    /// The <see cref="MLContext"/> is a starting point for all ML.NET operations. It is instantiated by user,
+    /// provides mechanisms for logging and entry points for training, prediction, model operations etc.
+    /// </summary>
+    public sealed class MLContext : IHostEnvironment
+    {
+        private readonly LocalEnvironment _env;
+
+        /// <summary>
+        /// Trainers and tasks specific to binary classification problems.
+        /// </summary>
+        public BinaryClassificationContext BinaryClassification { get; }
+        /// <summary>
+        /// Trainers and tasks specific to multiclass classification problems.
+        /// </summary>
+        public MulticlassClassificationContext MulticlassClassification { get; }
+        /// <summary>
+        /// Trainers and tasks specific to regression problems.
+        /// </summary>
+        public RegressionContext Regression { get; }
+        /// <summary>
+        /// Trainers and tasks specific to clustering problems.
+        /// </summary>
+        public ClusteringContext Clustering { get; }
+        /// <summary>
+        /// Trainers and tasks specific to ranking problems.
+        /// </summary>
+        public RankingContext Ranking { get; }
+
+        /// <summary>
+        /// Data processing operations.
+        /// </summary>
+        public TransformsCatalog Transform { get; }
```

Comment:
I know the other properties are not plural, but for some reason the singular `Transform` sticks out to me. It feels like it should be `Transforms`.

Reply:
Hmm, I like it the way it is. Maybe other commenters can swing the vote?

---
In reply to: [225254551](https://github.com/dotnet/machinelearning/pull/1252#discussion_r225254551) [](ancestors = 225254551)

Reply:
Ok, I renamed it

---
In reply to: [225262070](https://github.com/dotnet/machinelearning/pull/1252#discussion_r225262070) [](ancestors = 225262070,225254551)